### PR TITLE
allow _daemon_ user to run sudo

### PR DESCRIPTION
**Pull request by Thanhphan1147**
Allow the `_daemon_` user to run sudo without a password, used to allow agents to install packages depending on the use-case.

### Checklist

- [x] The [charm style guide](https://juju.is/docs/sdk/styleguide) was applied
- [x] The [contributing guide](https://github.com/canonical/is-charms-contributing-guide) was applied
- [x] The changes are compliant with [ISD054 - Managing Charm Complexity](https://discourse.charmhub.io/t/specification-isd014-managing-charm-complexity/11619)
- [x] The documentation is generated using `src-docs`
- [x] The documentation for charmhub is updated.
- [x] The PR is tagged with appropriate label (`urgent`, `trivial`, `complex`)

### CODE REVIEW

**Review by arturo-seijas — APPROVED** [review]

**Review by arturo-seijas — DISMISSED** [review]

**Review by yanksyoon — DISMISSED** [review]
LGTM! Could we include a small test to run sudo workflow?

**Review by arturo-seijas — DISMISSED** [review]

File: .trivyignore [hunk 1]
```diff
@@ -0,0 +1 @@
+CVE-2023-45288
```
Comment (arturo-seijas) [inline .trivyignore:1]: Missing blank line at EOF

**Review by arturo-seijas — APPROVED** [review]

File: .trivyignore (hunk 1 above)
Comment (Thanhphan1147) [reply .trivyignore]: updated!

**Review by yanksyoon — DISMISSED** [review]

**Review by yanksyoon — APPROVED** [review]

**Comment by Thanhphan1147** [comment]
> LGTM! Could we include a small test to run sudo workflow?

Good idea! Will include that in.

**Comment by Thanhphan1147** [comment]
Added an integration test for the sudo!

**Review by javierdelapuente — APPROVED** [review]

**Review by arturo-seijas — APPROVED** [review]

**Pull request merged**